js/ui/overlay/ui.overlay.js

```diff
@@ -502,12 +472,44 @@ const Overlay = Widget.inherit({
 
     _normalizePosition: function() {
         const position = this.option('position');
-        this._position = typeof position === 'function' ? position() : position;
+        const defaultPositionOptions = {
+            of: this.option('target'),
+            boundaryOffset: DEFAULT_BOUNDARY_OFFSET
+        };
+        if(isFunction(position)) {
+            this._position = extend(true, {}, defaultPositionOptions, position());
+        } else {
+            if(isDefined(position)) {
+                this._position = extend(true, {}, defaultPositionOptions, this._getPositionOptions(POSITION_ALIASES));
+            } else {
+                this._position = defaultPositionOptions;
+            }
+        }
     },
 
     _getAnimationConfig: function() {
         let animation = this.option('animation');
         if(isFunction(animation)) animation = animation.call(this);
+        const target = this.option('target');
+
+        if(isDefined(target)) {
```

Comment:
oh, I don't think it's a good idea to move `_initTarget` function code to this function.

`_initTarget` was called only in `initOptions` and `target` option changed handler.
But this function is called every time overlay shows or hides

Reply:
reworkerd